samcli/lib/providers/exceptions.py

```diff
@@ -83,3 +83,12 @@ def resource_identifier(self) -> "ResourceIdentifier":
     @property
     def property_name(self) -> str:
         return self._property_name
+
+
+class MissingFunctionNameException(Exception):
```

Comment:
Would this be a `UserException`, instead of a general `Exception`? This way it is returned as exit code `1` instead of `255`

Reply:
`UserException` is part of the interface and shouldn't be used in the `lib/`. This exception isn't propagated to the user, it's handled by the caller of invoke and a service response is chosen accordingly.